src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,356 @@
+
+        callerScriptOptionsMap.put("callerScriptPrefixLegalArgs", callerScriptPrefixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixArgValueTypes", callerScriptPrefixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPrefixMutexArgs", callerScriptPrefixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixAliasArgs", callerScriptPrefixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixMinOccurrences", callerScriptPrefixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPrefixMaxOccurrences", callerScriptPrefixMaxOccurrences);
+        if ( hasCallerScriptPrefixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "false");
+        }
+
+        callerScriptOptionsMap.put("callerScriptPostfixLegalArgs", callerScriptPostfixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixArgValueTypes", callerScriptPostfixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPostfixMutexArgs", callerScriptPostfixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixAliasArgs", callerScriptPostfixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixMinOccurrences", callerScriptPostfixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPostfixMaxOccurrences", callerScriptPostfixMaxOccurrences);
+        if ( hasCallerScriptPostfixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "false");
+        }
+
+        rootMap.put("callerScriptOptions", callerScriptOptionsMap);
+
+        // Get or create a template
+        final Template template = cfg.getTemplate(getIndexTemplateName());
+
+        // Create the output file
+        final File indexFile = new File(getDestinationDir(),
+                getIndexBaseFileName() + "." + getIndexFileExtension()
+        );
+
+        // Run the template and merge in the data
+        try (final FileOutputStream fileOutStream = new FileOutputStream(indexFile);
+             final OutputStreamWriter outWriter = new OutputStreamWriter(fileOutStream)) {
+            template.process(rootMap, outWriter);
+        } catch (TemplateException e) {
+            throw new DocException("Freemarker Template Exception during documentation index creation", e);
+        }
+    }
+
+    /**
+     * @return the name of the index template to be used for this doclet
+     */
+    @Override
+    public String getIndexTemplateName() { return "bash-completion.template.ftl"; }
+
+    /**
+     * @return The base filename for the index file associated with this doclet.
+     */
+    @Override
+    public String getIndexBaseFileName() { return callerScriptName + "-completion"; }
```

Comment:
There is two things that I do not fully understand about this, because I am not familiar with bash-completion:

1. If I want to use an uber-jar directly (through `java -jar Toolkit.jar`), is there any possibility to use bash-completion in that case? If it could be done, how this will affect the basename?
2. If the caller script is just a bash wrapper of `java -jar ToolKit.jar` named `toolkit.sh`, should the file be named `toolkit.sh-completion` or `toolkit-completion`?

Sorry for my ignorance, but I would like to know more about this and I think that it is a good opportunity to ask.